compiler/rustc_hir/src/hir.rs

```diff
@@ -2353,13 +2353,24 @@ pub struct TraitRef<'hir> {
     pub hir_ref_id: HirId,
 }
 
-impl TraitRef<'_> {
+impl TraitRef<'hir> {
+    /// Tries to create a new `TraitRef` object from the given path and `HirId`. Returns `None`
+    /// if the path resolved to something other than a trait or a trait alias.
```

Comment:
You need to mention how this handles `Res::Err`, since it's a bit surprising.

Reply:
Yeah - it's purely to not mess up the current tests, which as far as I can see from CI, was totally not how it turned out.